File: source/archive/plugin_src/ai/rorsch/pandagenie/nativelib/ArchiveLib.java
package ai.rorsch.pandagenie.nativelib;

public class ArchiveLib {

    static {
        System.loadLibrary("archive_module");
    }

    public native boolean compressZip(String[] inputPaths, String outputPath, String password);
    public native boolean decompressZip(String archivePath, String outputDir, String password);
    public native boolean compressTar(String[] inputPaths, String outputPath);
    public native boolean decompressTar(String archivePath, String outputDir);
    public native boolean compressGz(String inputPath, String outputPath);
    public native boolean decompressGz(String archivePath, String outputPath);
    public native boolean compressTarGz(String[] inputPaths, String outputPath);
    public native boolean decompressTarGz(String archivePath, String outputDir);
    public native String listContents(String archivePath);

    public static native void nativeConfigureSandbox(String[] readPaths, String[] writePaths);
    public static native void nativeClearSandbox();
}

File: source/archive/jni_bridge/ai/rorsch/pandagenie/nativelib/ArchiveLib.kt
package ai.rorsch.pandagenie.nativelib

class ArchiveLib {

    external fun compressZip(inputPaths: Array<String>, outputPath: String, password: String): Boolean
    external fun decompressZip(archivePath: String, outputDir: String, password: String): Boolean
    external fun compressTar(inputPaths: Array<String>, outputPath: String): Boolean
    external fun decompressTar(archivePath: String, outputDir: String): Boolean
    external fun compressGz(inputPath: String, outputPath: String): Boolean
    external fun decompressGz(archivePath: String, outputPath: String): Boolean
    external fun compressTarGz(inputPaths: Array<String>, outputPath: String): Boolean
    external fun decompressTarGz(archivePath: String, outputDir: String): Boolean
    external fun listContents(archivePath: String): String
}
